contracts/token/ERC20Rebasable.sol

```diff
@@ -0,0 +1,352 @@
+        _setERC20MetadataName(name_);
+        _setERC20MetadataSymbol(symbol_);
+    }
+
+    /// @inheritdoc IERC20Wrapable
+    function wrap(uint256 sharesAmount_) external returns (uint256) {
+        if (sharesAmount_ == 0) revert ErrorZeroSharesWrap();
+    
+        _mintShares(msg.sender, sharesAmount_);
+        if(!wrappedToken.transferFrom(msg.sender, address(this), sharesAmount_)) revert ErrorERC20Transfer();
+
+        return _getTokensByShares(sharesAmount_);
+    }
+
+    /// @inheritdoc IERC20Wrapable
+    function unwrap(uint256 tokenAmount_) external returns (uint256) {
+        if (tokenAmount_ == 0) revert ErrorZeroTokensUnwrap();
+
+        uint256 sharesAmount = _getSharesByTokens(tokenAmount_);
+        _burnShares(msg.sender, sharesAmount);
+        if(!wrappedToken.transfer(msg.sender, sharesAmount)) revert ErrorERC20Transfer();
+
+        return sharesAmount;
+    }
+
+    /// @inheritdoc IERC20Wrapable
+    function stETHPerToken() external view returns (uint256) {
+        return uint256(tokenRateOracle.latestAnswer());
+    }
+
+    /// @inheritdoc IERC20BridgedShares
+    function mintShares(address account_, uint256 amount_) external onlyBridge returns (uint256) {
+        return _mintShares(account_, amount_);
+    }
+
+    /// @inheritdoc IERC20BridgedShares
+    function burnShares(address account_, uint256 amount_) external onlyBridge {
+        _burnShares(account_, amount_);
+    }
+
+    /// @dev Validates that sender of the transaction is the bridge
+    modifier onlyBridge() {
+        if (msg.sender != bridge) {
+            revert ErrorNotBridge();
+        }
+        _;
+    }
+
+    /// @inheritdoc IERC20
+    function totalSupply() external view returns (uint256) {
+        return _getTokensByShares(totalShares);
+    }
+
+    /// @inheritdoc IERC20
+    function balanceOf(address account_) external view returns (uint256) {
+        return _getTokensByShares(_sharesOf(account_));
+    }
+
+    /// @inheritdoc IERC20
+    function approve(address spender_, uint256 amount_)
+        external
+        returns (bool)
+    {
+        _approve(msg.sender, spender_, amount_);
+        return true;
+    }
+
+    /// @inheritdoc IERC20
+    function transfer(address to_, uint256 amount_) external returns (bool) {
+        _transfer(msg.sender, to_, amount_);
+        return true;
+    }
+
+    /// @inheritdoc IERC20
+    function transferFrom(
+        address from_,
+        address to_,
+        uint256 amount_
+    ) external returns (bool) {
+        _spendAllowance(from_, msg.sender, amount_);
+        _transfer(from_, to_, amount_);
+        return true;
+    }
+
+    /// @notice Atomically increases the allowance granted to spender by the caller.
+    /// @param spender_ An address of the tokens spender
+    /// @param addedValue_ An amount to increase the allowance
+    function increaseAllowance(address spender_, uint256 addedValue_)
```

Comment:
next functions 
`function approve()` 
`function increaseAllowance()`
`function decreaseAllowance()` are allow zero to be used as the amount, is it ok?

Reply:
`approve()` for sure to allow zeroying the allowance ([reference](https://eips.ethereum.org/EIPS/eip-20#approve)
`increaseAllowance` and `decreaseAllowance` should be OK with zero amount at least because it is true for stETH

Reply:
Oh, I'm for removing them then